pkg/obs/specs/specs.go

```diff
@@ -0,0 +1,238 @@
+		if err != nil {
+			return nil, fmt.Errorf("user-supplied kubernetes version is not valid semver: %w", err)
+		}
+
+		kubeVersionString := kubeSemver.String()
+		kubeVersionParts := strings.Split(kubeVersionString, ".")
+
+		switch {
+		case len(kubeVersionParts) > 4:
+			logrus.Info("User-supplied Kubernetes version is a CI version")
+			logrus.Info("Setting channel to nightly")
+			pb.Channel = ChannelNightly
+		case len(kubeVersionParts) == 4:
+			logrus.Info("User-supplied Kubernetes version is a pre-release version")
+			logrus.Info("Setting channel to testing")
+			pb.Channel = ChannelTesting
+		default:
+			logrus.Info("User-supplied Kubernetes version is a release version")
+			logrus.Info("Setting channel to release")
+			pb.Channel = ChannelRelease
+		}
+	}
+
+	// This function determines the Kubernetes version if it's not provided.
+	pb.KubernetesVersion, err = c.GetKubernetesVersion(pb.KubernetesVersion, pb.Channel)
+	if err != nil {
+		return nil, fmt.Errorf("getting kubernetes version: %w", err)
+	}
+
+	logrus.Infof("Using Kubernetes version %s", pb.KubernetesVersion)
+
+	// This function gets download link base depending on what channel are
+	// we using.
+	pb.DownloadLinkBase, err = c.GetDownloadLinkBase(pb.KubernetesVersion, pb.Channel)
+	if err != nil {
+		return nil, fmt.Errorf("getting kubernetes download link base: %w", err)
+	}
+
+	logrus.Infof("Kubernetes download link base: %s", pb.DownloadLinkBase)
+
+	// For cases where a CI build version of Kubernetes is retrieved, replace instances
+	// of "+" with "-", so that we build with a valid Debian package version.
+	pb.KubernetesVersion = strings.Replace(pb.KubernetesVersion, "+", "-", 1)
+
+	logrus.Infof("Successfully constructed package builder!")
+
+	return pb, nil
+}
+
+// ConstructPackageDefinitions creates instances of PackageDefinition based
+// on what packages we selected to build.
+func (c *Client) ConstructPackageDefinitions(pkgBuilder *PackageBuilder) error {
+	logrus.Infof("Constructing package definitions...")
+	var err error
+
+	if pkgBuilder == nil {
+		return errors.New("package builder cannot be nil")
+	}
+
+	for _, pkg := range c.options.Packages {
+		pkgDef := &PackageDefinition{
+			Name:         pkg,
+			Revision:     c.options.Revision,
+			Dependencies: map[string]string{},
+		}
+
+		// Determine the package version.
+		pkgDef.Version, err = c.GetPackageVersion(pkgDef, pkgBuilder.KubernetesVersion)
+		if err != nil {
+			return fmt.Errorf("getting package %q version: %w", pkg, err)
+		}
+
+		logrus.Infof("%s package version: %s", pkgDef.Name, pkgDef.Version)
+
+		// Determine dependencies for given package.
+		pkgDef.Dependencies, err = GetDependencies(pkgDef)
+		if err != nil {
+			return fmt.Errorf("getting dependencies for %q: %w", pkg, err)
+		}
+
+		pkgBuilder.Definitions = append(pkgBuilder.Definitions, pkgDef)
+	}
+
+	logrus.Infof("Successfully constructed package definitions!")
```

Comment:
ultra nit, we can use 
```suggestion
	logrus.Info("Successfully constructed package definitions!")
```
if no format is included.